plugins/scale/scale-title-overlay.cpp

```diff
@@ -0,0 +1,296 @@
+        overlay.render_text(view->get_title().c_str(), par);
+        view->damage();
+    }
+
+  public:
+    enum class position
+    {
+        TOP,
+        CENTER,
+        BOTTOM,
+    };
+
+    position pos = position::CENTER;
+
+    GLuint get_texture(float output_scale, wf::geometry_t& geometry)
+    {
+        if (output_scale != par.output_scale)
+        {
+            par.output_scale = output_scale;
+            render_overlay();
+        }
+
+        wlr_box box = get_transformed_bounding_box();
+        int w = overlay.tex.width;
+        int h = overlay.tex.height;
+        int y = 0;
+        switch (pos)
+        {
+          /* TODO: Add some padding! */
+          case position::TOP:
+            y = box.y - (int)(h / output_scale);
+            break;
+
+          case position::CENTER:
+            y = box.y + box.height / 2 - (int)(h / output_scale / 2);
+            break;
+
+          case position::BOTTOM:
+            y = box.y + box.height;
+            break;
+        }
+
+        geometry = {box.x + box.width / 2 - (int)(w / output_scale / 2),
+            y, (int)(w / output_scale), (int)(h / output_scale)};
+        return overlay.tex.tex;
+    }
+
+    explicit view_title_overlay(wayfire_view v, wf::scale_transformer& tr_,
+        int font_size,
+        const wf::color_t& bg_color, const wf::color_t& text_color) : view(v),
+        tr(tr_)
+    {
+        par.font_size  = font_size;
+        par.bg_color   = bg_color;
+        par.text_color = text_color;
+        par.exact_size = true;
+        render_overlay();
+        view->connect_signal("title-changed", &view_changed);
+        view->connect_signal("geometry-changed", &view_changed);
+    }
+
+    wf::signal_connection_t view_changed{[this] (auto)
+        {
+            render_overlay();
+        }
+    };
+
+    wf::dimensions_t get_size() const
+    {
+        return {overlay.tex.width, overlay.tex.height};
+    }
+};
+
+
+scale_show_title::scale_show_title() :
+    view_filter{[this] (wf::signal_data_t *data)
+    {
+        update_title_overlay_opt();
+
+        auto signal = static_cast<scale_filter_signal*>(data);
+        const std::string& pos = title_position;
+        if (pos == "center")
+        {
+            return;
+        }
+
+        unsigned int extra_pad = wf::cairo_text_t::measure_height(
+            title_font_size, true);
+        if (pos == "top")
+        {
+            signal->pad.top = std::max(signal->pad.top, extra_pad);
+        } else if (pos == "bottom")
+        {
+            signal->pad.bottom = std::max(signal->pad.bottom, extra_pad);
+        }
+    }},
+
+    scale_end{[this] (wf::signal_data_t*)
+    {
+        show_view_title_overlay = title_overlay_t::NEVER;
+        last_title_overlay = nullptr;
+        if (mouse_update_connected)
+        {
+            wf::get_core().disconnect_signal(&mouse_update);
+            mouse_update_connected = false;
+        }
+    }
+},
```

Comment:
I could not get uncrustify to be consistent with this formatting (all of these are part of the initializer list).

Reply:
alright, I'll push a fix when the pr is ready for merging